### DISCUSSION TITLE
String Handling Functions Planned?

### DISCUSSION THREAD

**Discussion by fpuhan** [benkeen/generatedata#787] [Ideas]
In an earlier release, there was a wealth of string-handling capabilities built in, such as TRUNCATE, LOWER, etc.  I used this to create computed email addresses:  {$ROW1|TRUNCATE:1:""|LOWER}{$ROW2|LOWER}@email.com which would take the first character (initial) from ROW1, concatenate it with ROW2 and append "@email.com."  I don't see this feature in the current release.  Is this planned for inclusion?

**Comment by benkeen** [comment]
Yes, it was a good feature, hey? But it was all a part of using Smarty as the template language for the PHP app - that provided all that functionality out the box. When we moved to node, react, redux, TS etc. etc. we lost all that. 

However, the Computed Data Type does let you enter arbitrary JS. For example:

- Lowercase: `{{ROW1.toLowerCase()}}`
- Truncate to one char: `{{ROW1.substring(0, 1)}}`

So your example: `{{ROW1.toLowerCase().substring(0, 1)}}{{ROW2.toLowerCase()}}@email.com`

It's not as friendly as the Smarty syntax, but still doable!

**Comment by fpuhan** [comment]
Hey Ben,

Thanks for the reply and info on how to accomplish the task I was trying to do!   Actually, the functions look easier to me, not harder.

Incidentally, I had some problems with the Postal/Zip data.  The preview, even when matched to the region field, came back often with six characters (not five, as I specified US) and sometimes mixed alphanumeric characters.  Well, I thought I’d go ahead and generate some data and process never got past 0.  I tried a number of different settings, but each time the Postal/Zip data borked the generation.  Removing the field solved the problem (I might just try using a random string of five numeric characters, but I thought I’d mention it. 

I really like the service.  If I had more need for it than once every few years, I’d subscribe to it right away!

Fred

**Comment by benkeen** [comment]
> Incidentally, I had some problems with the Postal/Zip data.  The preview, even when matched to the region field, came back often with six characters (not five, as I specified US) and sometimes mixed alphanumeric characters.  Well, I thought I’d go ahead and generate some data and process never got past 0.  I tried a number of different settings, but each time the Postal/Zip data borked the generation.  Removing the field solved the problem (I might just try using a random string of five numeric characters, but I thought I’d mention it.

Mmm, thanks! Maybe a bug. I'll take a look and see if I can reproduce.

**Comment by fpuhan** [comment]
Hi Ben,

It worked!  Not only did I get the email formatting the way I wanted, but now the Postal/Zip code appears to be functioning properly.  I don’t know if you did something between our last correspondence, but I did make a change to my layout:  I chose Alphanuermic as a data type and US Zip Code (90210 - cute!) as the format and while I’m no expert on ZIP codes, the few I know appear to line up with the region correctly.

Thanks for this awesome tool!

Fred
